packages/python-sdk/e2b/sandbox_sync/sandbox_api.py

```diff
@@ -19,36 +19,59 @@
 )
 from e2b.api.client.models import (
     NewSandbox,
+    SandboxState,
     PostSandboxesSandboxIDTimeoutBody,
     ResumedSandbox,
 )
+from e2b.api.client.types import UNSET
 from e2b.connection_config import ConnectionConfig
 from e2b.exceptions import NotFoundException, TemplateException
 from e2b.sandbox.sandbox_api import SandboxApiBase, SandboxInfo, SandboxMetrics
 from httpx import HTTPTransport
 from packaging.version import Version
 
+class ListSandboxesResponse:
+     def __init__(self, sandboxes: List[SandboxInfo], has_more_items: bool, next_token: Optional[str], iterator: Generator[SandboxInfo, None, None]):
+         self.sandboxes = sandboxes
+         self.has_more_items = has_more_items
+         self.next_token = next_token
+         self.iterator = iterator
+
 
 class SandboxApi(SandboxApiBase):
+    @dataclass
+    class SandboxQuery:
+        """Query parameters for listing sandboxes."""
+
+        metadata: Optional[dict[str, str]] = None
+        """Filter sandboxes by metadata."""
+
+        state: Optional[List[SandboxState]] = None
+        """Filter sandboxes by state."""
```

Comment:
Not sure about this, it's nice that you don't have to import anything, but you have to always write `Sandbox.SandboxQuery` instead of `SandboxQuery` if you import it and also now it lives on the `Sandbox` object, which makes it less readable, I would try to keep the number of attributes on `Sandbox` object to a minimum.